src/objects.rs

```diff
@@ -691,6 +691,44 @@ impl GetObjectType for VehicleJourney {
     }
 }
 
+#[derive(Debug)]
+pub enum StopTimeError {
+    DuplicateStopSequence { vj_id: String },
+    IncoherentStopTimes { vj_id: String },
+}
```

Comment:
Excellent idea to introduce a specific type to characterize the error. However, a few remarks.

# Give more information in the error variants

Since you define different variants for the different types, you can use them to keep more information about the error itself, so it would help whoever is trying to make sense of it.

For example, this could be done.

```suggestion
pub enum StopTimeError {
    DuplicateStopSequence {
    	vj_id: String,
    	duplicated_sequence: u32,
    },
    IncoherentStopTimes {
    	vj_id: String,
    	first_incorrect_sequence: u32,
    	first_incorrect_time: Time,
    },
}
```

# Implements `std::error::Error`

Since you're defining an error type, let's implement the standard trait for errors in Rust. Since you're would implement `std::error::Error` and you're implementing `std::fmt::Display` and `std::fmt::Debug`, it would be an excellent opportunity to bring the crate `thiserror` for this.

```suggestion
#[derive(thiserror::Error)]
pub enum StopTimeError {
    #[error("duplicate stop_sequence '{duplicated_sequence}' for the trip '{vj_id}'")]
    DuplicateStopSequence {
    	vj_id: String,
    	duplicated_sequence: u32,
    },
    #[error("incoherent stop times '{first_incorrect_sequence}' at time '{first_incorrect_time}' for the trip '{vj_id}'")]
    IncoherentStopTimes {
    	vj_id: String,
    	first_incorrect_sequence: u32,
    	first_incorrect_time: Time,
    },
}
```

Reply:
done in d03225efc5decf59f273b03e368313e49e03f244